lib/AsymmetricKey.js

```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
     return JSON.parse(JSON.stringify(this._keyDescription));
   }
 
+  /**
+   * Gets an algorithm from a key description.
+   *
+   * @param {object} options - The options to use.
+   * @param {object} options.keyDescription - A key description.
+   *
+   * @returns {string} The resulting algorithm.
+   */
+  getAlgorithm({keyDescription}) {
+    const {publicKeyMultibase} = keyDescription;
+    // supported algorithms P-256, P-384, P-521, ed25519, secp256k1, P-256K
+    // @see https://github.com/multiformats/multicodec/blob/master/table.csv
+    // maps the bs58 prefix to the JOSE algorithm name
+    const algorithms = new Map([
+      ['6Mk', 'Ed25519'],
+      ['Dna', 'P-256'],
+      ['82L', 'P-384'],
+      ['2J9', 'P-521'],
+      // FIXME this prefix was not generated by an ed448 did key implementation
+      ['6GP', 'Ed448'],
+      ['Q3s', 'secp256k1']
+    ]);
```

Comment:
This should be a constant external to this function (there's no need to recreate it on every function call).

Reply:
Addressed here: https://github.com/digitalbazaar/webkms-client/pull/42/commits/b2b6f51cb5738d003a1b2b89ea4a2f6192cd397d , but do we want a `constats.js` module to put this in as we might want this functionality in other key classes?